tests/unit/v2/server/testController.js

```diff
@@ -0,0 +1,113 @@
+const assert = require('assert');
+
+const APIController = require('../../../../libV2/server/controller');
+const healthcheck = require('../../../../libV2/server/API/internal/healthcheck');
+
+const { ResponseContainer } = require('../../../../libV2/models');
+const { ExpressResponseStub, templateRequest } = require('../../../utils/v2Data');
+
+
+describe('Test APIController', () => {
+    it('should load a valid module', () => {
+        const module = APIController._safeRequire('./controller');
+        assert.deepStrictEqual(module, APIController);
+    });
+
+    it('should return null for a MODULE_NOT_FOUND error', () => {
+        const module = APIController._safeRequire('/tmp/doesntexist');
+        assert.deepStrictEqual(module, null);
+    });
+
+    it('should rethrow a non-MODULE_NOT_FOUND error', () => {
+        // We use a json file to avoid mocha trying to load it and exiting
+        assert.throws(() => APIController._safeRequire('../../tests/unit/v2/server/hasanerror.json'));
+    });
+
+    it('should load an operation handler', () => {
+        const handler = APIController._getOperationHandler('internal', 'healthcheck');
+        assert(typeof handler === 'function');
+    });
+
+    it('should patch a NotImplemented handler for MODULE_NOT_FOUND errors', () => {
+        const handler = APIController._getOperationHandler('invalid', 'idontexist');
+        assert(typeof handler === 'function');
+    });
+
+    it('should load handlers for a tag', () => {
+        const handlers = APIController._collectHandlers('internal');
+        assert.deepStrictEqual(handlers, {
+            healthcheck,
+        });
+    });
+
+    it('should extract a requests parameters', () => {
+        const req = {
+            swagger: {
+                params: {
+                    foo: {
+                        value: 'bar',
+                    },
+                },
+            },
+        };
+        assert(APIController._extractParams(req), { foo: 'bar' });
```

Comment:
did you mean to assert deep equals here? This just checks _extractParams returns a truthy value I believe